dulwich/tests/compat/test_repository.py

```diff
@@ -119,3 +127,32 @@ def test_packed_objects(self):
     def test_all_objects(self):
         expected_shas = self._get_all_shas()
         self.assertShasMatch(expected_shas, iter(self._repo.object_store))
+
+
+@skipIf(git_version() < (2, 5), 'Git version must be >= 2.5')
+class WorkingTreeTestCase(ObjectStoreTestCase):
```

Comment:
This doesn't have anything to do with ObjectStore, so please derive from CompatTestCase